### PR TITLE
Add better PlayerVelocityEvent support

### PR DESCRIPTION
**Pull request by Archonic944**
My PRQ allows API users to check if a PlayerVelocityEvent was caused by the player being attacked by another player. This will pave the way for better and more direct knockback manipulation.

Please provide a detailed description of the changes you have made.

Adds a final boolean variable in PlayerVelocityEvent that defines if the event was caused by a player being attacked by another player.

# How has this been tested?
As I simply just added a variable inside the PlayerVelocityEvent class that overflows with the default constructor, I have not tested this change.
# Checklist:

- [X] I have reviewed my code thoroughly.
- [ ] I have tested my code.
- [X] My changes generate no new warnings

### CODE REVIEW

**Review by LewUwU — CHANGES_REQUESTED** [review]
change fromPlayerBeingAttacked to fromAttack

**Comment by Archonic944** [comment]
> change fromPlayerBeingAttacked to fromAttack

Done!

Are we allowed to ping someone with the power to merge this PRQ?

File: NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java [hunk 1]
```diff
@@ -541,6 +542,17 @@ public void setPullWhileLeashed(boolean pullWhileLeashed) {
     }
     // TacoSpigot end
 
+    // Nacho start
+    public void heal(float amount){
+        if(getHealth() + amount <= getMaxHealth()) getHandle().heal(amount);
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java:547]:
```suggestion
        if(getHealth() + amount <= getMaxHealth()) getHandle().heal(amount);
        else getHandle().setHealth(getMaxHealth());
```

**Review by Elierrr — CHANGES_REQUESTED** [review]

File: NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java [hunk 2]
```diff
@@ -541,6 +542,17 @@ public void setPullWhileLeashed(boolean pullWhileLeashed) {
...
+    }
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java:548]:
```suggestion
    public void heal(float amount){
        double max = getMaxHealth()-getHealth();
        if ((health < 0) || (health > max)) {
            throw new IllegalArgumentException("Heal amount must be between 0 and " + max + ", but was " + health
                + ". (attribute base value: " + this.getHandle().getAttributeInstance(GenericAttributes.maxHealth).b()
            + (this instanceof CraftPlayer ? ", player: " + this.getName() + ')' : ')'));
        }

        getHandle().heal(amount);
    }
```

File: NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java [hunk 3]
```diff
@@ -541,6 +542,17 @@ public void setPullWhileLeashed(boolean pullWhileLeashed) {
...
+
+    public void heal(float amount, EntityRegainHealthEvent.RegainReason reason){
+        if(getHealth() + amount <= getMaxHealth())
+            getHandle().heal(amount, reason);
+    }
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java:553]:
```suggestion
    public void heal(float amount, EntityRegainHealthEvent.RegainReason reason){
        double max = getMaxHealth()-getHealth();
        if ((health < 0) || (health > max)) {
            throw new IllegalArgumentException("Heal amount must be between 0 and " + max + ", but was " + health
                + ". (attribute base value: " + this.getHandle().getAttributeInstance(GenericAttributes.maxHealth).b()
            + (this instanceof CraftPlayer ? ", player: " + this.getName() + ')' : ')'));
        }
        getHandle().heal(amount, reason);
    }
```

**Review by Elierrr — CHANGES_REQUESTED** [review]

File: NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java (hunk 3 above)
Comment (Archonic944) [reply NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java:553]: Uhh, that part of my PRQ is already merged. I'm not sure how to separate these changes into a new one, though.

**Pull request closed without merging**